spec/models/program_spec.rb

```diff
@@ -241,6 +241,48 @@
     end
   end
 
+  describe '#any_event_for_this_date?' do
+
+    let(:event){ create(:event, program: program) }
+
+    context 'when no schedule is selected for the conference' do
+      let(:schedule) { create(:schedule, program: program) }
+      let!(:event_schedule) { create(:event_schedule, event: event, schedule: schedule, start_time: DateTime.parse("#{Date.current + 1} 10:00").utc) }
+
+      it 'returns false irrespective of any date' do
+        expect(program.any_event_for_this_date?(Date.current + 1)).to eq false
+      end
+
+      it 'returns false if date passed is empty' do
+        expect(program.any_event_for_this_date?('')).to eq false
+      end
+
+      it 'returns false if date passed is nil' do
+        expect(program.any_event_for_this_date?(nil)).to eq false
+      end
+    end
+
+    context 'when schedule is selected for the conference' do
```

Comment:
And where is the selected schedule for the conference defined?

Reply:
It is defined in event_schedule factory

Reply:
Check https://github.com/openSUSE/osem/blob/master/spec/factories/event_schedule.rb#L17

Reply:
I am inclined to explicitly define a schedule in a before block, after the let definition, because this is the very thing we are testing here (what happens when there is a schedule).